tests/compat.sh

```diff
@@ -810,6 +800,10 @@ run_client() {
     LEN=$(( 72 - `echo "$TITLE" | wc -c` ))
     for i in `seq 1 $LEN`; do printf '.'; done; printf ' '
 
+    # Translate ciphersuite names based on client's naming convention
+    t_cipher=$(./scripts/translate_ciphers.py $3 $2)
```

Comment:
About calling `translate_ciphers` more times: the performance is a concern. Calling a Python script isn't cheap, but how expensive is it here? Can you please compare the run times for a sample run in the old and new versions?

Ideally we'd call `translate_ciphers.py` only once and populate an associative array. But this requires bash 4. In practice everyone has bash, but not everyone has bash 4 because macOS only comes with bash 3 (you have to get a modern bash from elsewhere, e.g. brew, and I don't know if everyone does that).

Alternatively, it's possible to simulate associative arrays (enough for our purposes here) with individual variables and `eval`. It's not pretty, but the ugly stuff would be in just one auxiliary function with a clear interface. It requires `_` rather than `-` in the key names.

Alternatively, since both keys and values use a restricted alphabet, we can construct a giant string like `TLS-OFFICIAL-FOO:openssl:OPENSSL-FOO TLS-OFFICIAL-FOO:gnutls:GNUTLS-FOO ...` and use basic string manipulation to find the desired entry. Again, not pretty, but the ugly stuff would be in just one auxiliary function. The string would be long, but at a guess not so long that it would be a noticeable performance hit.

Reply:
Attach a simple performance testing:

Execution time in old versions:

```
$ ./tests/compat.sh
…………
PASSED (1428 / 1428 tests (2 skipped))
execution time was 251 s.
```
Execution time in new versions:
```
$ ./tests/compat.sh
…………
PASSED (1428 / 1428 tests (2 skipped))
execution time was 331 s.
```

This is tested in my local workstation. It looks the execution time is increased by 30%. I don't have much idea of how this performance decrease impacts our CI. What's your opinion?

I think simulating associative arrays is a very good option although I haven't had a clear outline of code structure. If the performance matters, I'm happy with the provided solution.

Reply:
Thanks for doing the benchmark! 30% slower is a lot, it makes me feel the interface improvement isn't really worth it. I'm familiar with sh/bash so I'll try out some other potential solutions next week.

Reply:
As discussed, I looked at how to go back to calling `translate_ciphers.py` once per server start, rather than once per client start. I made a simple tweak to the output format of `translate_ciphers.py`. It's not pretty, but it works. While I was at it I made other tweaks to speed up the inner loop a little (the effect is measurable but a lot less than the reduction of calls to the python script).

Please review my commits.

Reply:
Thanks for providing a better solution to speed up `compat.sh`. Overall the commits look to me. Then I made a simple performance testing and got below results (in the same test environment with single run). 

Execution time with head of 449bd83
```
$ ./tests/compat.sh
…………
PASSED (1428 / 1428 tests (2 skipped))
execution time was 261 s.
```

Execution time with head of 60f8eaa
```
$ ./tests/compat.sh
…………
PASSED (1428 / 1428 tests (2 skipped))
execution time was 330 s.
```

Execution time with head of b20028b
```
$ ./tests/compat.sh
…………
PASSED (1428 / 1428 tests (2 skipped))
execution time was 237 s.
```

So test speed is increased by 10%. I believe this is a good improvement.